aiida_vasp/workchains/vasp.py

```diff
@@ -273,3 +297,95 @@ def on_except(self, exc_info):
                         'Please inspect messages and act.')
 
         return super(VaspWorkChain, self).on_except(exc_info)
+
+    @process_handler(process_handler=1000)
+    def _handle_misc_exists(self, node):
+        # Check if the run is converged electronically
+        if 'misc' not in node.outputs:
+            self.report('Cannot found `misc` outputs.')
+            return ProcessHandlerReport(exit_code=self.exit_codes.ERROR_UNKNOWN)
+        return None
+
+    @process_handler(priority=0)
+    def _handle_calculation_sanity_checks(self, node):  # pylint: disable=no-self-use,unused-argument
+        """
+        Perform additional sanity checks on successfully completed calculation.
+
+        Calculations that run successfully may still have problems that can only be determined when inspecting
+        the output. The same problems may also be the hidden root of a calculation failure. For that reason,
+        after verifying that the calculation ran, regardless of its calculation state, we perform some sanity
+        checks. However, make sure that this workchain is general. Calculation plugin specific sanity checks
+        should go into the childs..
+
+        Note that this is the "final" check, most of the potential problems should have been handled and
+        fixed in restart by error handlers.
+        """
+
+        misc = node.outputs.misc.get_dict()
+        if 'run_status' not in 'misc':
```

Comment:
> The plugin and parser checks are at the calculation level and here it is at the work chain level. So they are not equivalent actually.

Yes, see my second comment. In fact first check is at the parser level.

> This handler is necessary here as the user may want to disable the automatic handling of electronic /ionic convergence failures. In that case, this handler would throw the corresponding exit_code. Otherwise, the non-zero exit code would be left unhandled and the calculation will be restarted for one more times, which is the default behaviour.

The comment is not about the handler, which is fine and necessary, but the check of the `run_status`. This is already done in the parser (but yes, they are different). The only reason for checking here is to detect errors in the node composers etc. (which we should catch in development with tests) or if the user can disable the `misc` output node, or the quantities in it at the workchain level. In fact, what we should do is to tailor the required output at the workchain level. So if the user want the handlers, the workchain should check that we have the correct `misc` node configured in the parser. Then there is one case left, the user disables the `misc` node and the handlers. But then we would never enter this method?